pandas/core/series.py

```diff
@@ -872,6 +872,7 @@ def __getitem__(self, key):
 
             if com.is_bool_indexer(key):
                 key = check_bool_indexer(self.index, key)
+                key = np.asarray(key, dtype=bool)
```

Comment:
doesn’t check_bool_indexer guarantee this?

Reply:
i was surprised by this too, but no

Reply:
ok, let's investigate if we can do this *inside* check_bool_indexer (either open an issue or do as a followon)